CMake build fails due to `Unknown CMake command "mbed_set_mbed_target_linker_script"`

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   PLEASE ENSURE ALL SECTIONS OF THIS TEMPLATE ARE FILLED IN AND THAT THERE ARE
   NO OTHER CHANGES TO THE TEMPLATE.    
   
   Only bugs should be raised here as issues. Questions or enhancements should instead be raised on 
   our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

Error Message

Failed when executing: python -u ../mbed-os/tools/test/examples/examples.py  compile GCC_ARM --mcu NUCLEO_F767ZI --cmake -v   Reason: hudson.AbortException: script returned exit code 1

Standard Output

Last 100 lines from the log:
Total Flash memory (text + data): 33624(+33624) bytes


#### STDERR ####

#### End of STDOUT/STDERR ####

[EXAMPLES]> WARNING  Test for mbed-os-example-blinky will not be generated.
[EXAMPLES]> INFO     In folder 'mbed-os-example-blinky-baremetal'
[EXAMPLES]> INFO     In folder 'mbed-os-example-tls/benchmark'
[EXAMPLES]> INFO     In folder 'mbed-os-example-tls/tls-client'
[EXAMPLES]> INFO     In folder 'mbed-os-example-tls/hashing'
[EXAMPLES]> INFO     In folder 'mbed-os-example-tls/authcrypt'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_Advertising'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GAP'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GattClient_CharacteristicUpdates'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GattClient_CharacteristicWrite'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GattServer_AddService'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GattServer_CharacteristicUpdates'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_GattServer_CharacteristicWrite'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_PeriodicAdvertising'
[EXAMPLES]> INFO     In folder 'mbed-os-example-ble/BLE_SecurityAndPrivacy'
[EXAMPLES]> INFO     In folder 'mbed-os-example-sockets'
[EXAMPLES]> INFO     In folder 'mbed-os-example-cellular'
[EXAMPLES]> INFO     In folder 'mbed-os-example-lorawan'
[EXAMPLES]> INFO     In folder 'mbed-os-example-devicekey'
[EXAMPLES]> INFO     In folder 'mbed-os-example-thread-statistics'
[EXAMPLES]> INFO     In folder 'mbed-os-example-sys-info'
[EXAMPLES]> INFO     In folder 'mbed-os-example-cpu-stats'
[EXAMPLES]> INFO     In folder 'mbed-os-example-socket-stats'
[EXAMPLES]> INFO     In folder 'mbed-os-example-error-handling'
[EXAMPLES]> INFO     In folder 'mbed-os-example-filesystem'
[EXAMPLES]> INFO     In folder 'mbed-os-example-mesh-minimal'
[EXAMPLES]> INFO     Compiling mbed-os-example-mesh-minimal NUCLEO_F767ZI GCC_ARM
[EXAMPLES]> INFO     Executing command 'mbed-tools compile -t GCC_ARM -m NUCLEO_F767ZI -c'...

#### STDOUT ####
Configuring project and generating build system...
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.7 (found version "3.7.4") found components: Interpreter 
-- Checking for Python package prettytable -- found
-- Checking for Python package future -- found
-- Checking for Python package jinja2 -- found
-- Checking for Python package intelhex -- found
-- Configuring incomplete, errors occurred!
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@2/examples/mbed-os-example-mesh-minimal/cmake_build/NUCLEO_F767ZI/develop/GCC_ARM/CMakeFiles/CMakeOutput.log".
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@2/examples/mbed-os-example-mesh-minimal/cmake_build/NUCLEO_F767ZI/develop/GCC_ARM/CMakeFiles/CMakeError.log".

#### STDERR ####
CMake Error at CMakeLists.txt:18 (mbed_set_mbed_target_linker_script):
  Unknown CMake command "mbed_set_mbed_target_linker_script".


ERROR: CMake invocation failed!

More information may be available by using the command line option '-v'.

#### End of STDOUT/STDERR ####

[EXAMPLES]> INFO     In folder 'mbed-os-example-bootloader'
[EXAMPLES]> INFO     In folder 'mbed-os-example-mbed-crypto/getting-started'
[EXAMPLES]> INFO     In folder 'mbed-os-example-psa'
[EXAMPLES]> INFO     In folder 'mbed-os-example-nfc/NFC_EEPROM'
[EXAMPLES]> INFO     In folder 'mbed-os-example-nfc/NFC_SmartPoster'
[EXAMPLES]> INFO     In folder 'mbed-os-example-blockdevice'
[EXAMPLES]> INFO     In folder 'mbed-os-example-kvstore'
[EXAMPLES]> INFO     In folder 'mbed-os-example-crash-reporting'
[EXAMPLES]> INFO     In folder 'mbed-os-example-sd-driver'
[EXAMPLES]> INFO     In folder 'mbed-os-example-attestation'
[EXAMPLES]> INFO     In folder 'mbed-os-example-atecc608a/atecc608a'


Passed Example Compilation:
+------------------------+---------------+-----------+----------+--------------+
| EXAMPLE NAME           |     TARGET    | TOOLCHAIN | TEST GEN | BUILD RESULT |
+------------------------+---------------+-----------+----------+--------------+
| mbed-os-example-blinky | NUCLEO_F767ZI |  GCC_ARM  | TEST_OFF |    PASSED    |
+------------------------+---------------+-----------+----------+--------------+


Failed Example Compilation:
+------------------------------+---------------+-----------+----------+--------------+
| EXAMPLE NAME                 |     TARGET    | TOOLCHAIN | TEST GEN | BUILD RESULT |
+------------------------------+---------------+-----------+----------+--------------+
| mbed-os-example-mesh-minimal | NUCLEO_F767ZI |  GCC_ARM  |  UNSET   |    FAILED    |
+------------------------------+---------------+-----------+----------+--------------+
Number of failures = 1


#### Target(s) affected by this defect ?

All

#### Toolchain(s) (name and version) displaying this defect ?

GCC_ARM, ARM

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in:
    https://github.com/ARMmbed/mbed-os/blob/master/platform/include/platform/mbed_version.h
    The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
     
-->

mbed-os-0.0.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Mbed CLI 2 7.8.0

#### How is this defect reproduced ?

`mbed-tools compile -t GCC_ARM -m NUCLEO_F767ZI -c`

--- a/.travis.yml
+++ b/.travis.yml
@@ -79,9 +79,9 @@ matrix:
         - pip install "Jinja2>=2.10.1,<2.11"
         - pip install "intelhex>=1.3,<=2.2.1"
       script:
-        - mbedtools checkout
-        - echo mbedtools build -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
-        - mbedtools build -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
+        - mbedtools deploy
+        - echo mbedtools compile -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
+        - mbedtools compile -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
         - ccache -s
 
     - <<: *cmake-build-test

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,10 +13,6 @@ add_subdirectory(${MBED_PATH})
 
 add_executable(${APP_TARGET})
 
-mbed_configure_app_target(${APP_TARGET})
-
-mbed_set_mbed_target_linker_script(${APP_TARGET})
-
 project(${APP_TARGET})
 
 target_include_directories(${APP_TARGET}

--- a/README.md
+++ b/README.md
@@ -8,14 +8,22 @@ The application can be built for the unsecure 6LoWPAN-ND or Wi-SUN network.
 
 See the [6LoWPAN overview](https://os.mbed.com/docs/latest/reference/mesh-tech.html) for the definition of star and mesh networks. These same principles apply also to Wi-SUN protocol.
 
+
+## Mbed OS build tools
+
+### Mbed CLI 2
+Starting with version 6.5, Mbed OS uses Mbed CLI 2. It uses Ninja as a build system, and CMake to generate the build environment and manage the build process in a compiler-independent manner. If you are working with Mbed OS version prior to 6.5 then check the section [Mbed CLI 1](#mbed-cli-1).
+1. [Install Mbed CLI 2](https://os.mbed.com/docs/mbed-os/latest/build-tools/install-or-upgrade.html).
+1. From the command-line, import the example: `mbed-tools import mbed-os-example-mesh-minimal`
+1. Change the current directory to where the project was imported.
+
+### Mbed CLI 1
+1. [Install Mbed CLI 1](https://os.mbed.com/docs/mbed-os/latest/quick-start/offline-with-mbed-cli.html).
+1. From the command-line, import the example: `mbed import mbed-os-example-mesh-minimal`
+1. Change the current directory to where the project was imported.
+
+
 ## Setup
-
-### Download the application
-
-```
-mbed import mbed-os-example-mesh-minimal
-cd mbed-os-example-mesh-minimal
-```
 
 ### Change the channel settings (optional)
 
@@ -119,21 +127,9 @@ To use the S2LP, modify the `mbed_app.json` file by setting:
 
 Note that the provide-default for each radio driver defaults to false, so there is no need to list all the possible drivers in the `mbed_app.json`.
 
-### Compile the application
-
-```
-mbed compile -m K64F -t GCC_ARM
-```
-
-A binary is generated in the end of the build process.
-
 ### Connect the RF shield to the board
 
 We are using the Atmel AT86RF233, which you can [purchase](https://l-tek.si/web-shop/l-tek-6lowpan-arduino-shield-2-4ghz/). Place the shield on top of your board, and power it up.
-
-### Program the target
-
-Drag and drop the binary to the target to program the application.
 
 ### Update the firmware of the border router
 
@@ -144,6 +140,25 @@ This example supports the following border router:
 The border router supports [static and dynamic backhaul configuration](https://github.com/ARMmbed/nanostack-border-router#the-backhaul-configuration-options). The static configuration is good for testing, but the dynamic one works if your network infrastructure is supplying an IPv6 address. Make sure that you use the appropiate mode.
 
 Remember to connect the Ethernet cable between the border router and your home/office router. Then power on the board.
+
+## Building and running
+
+1. Connect a USB cable between the USB port on the board and the host computer.
+1. Run the following command to build the example project and program the microcontroller flash memory:
+
+    * Mbed CLI 2
+
+    ```bash
+    $ mbed-tools compile -m <TARGET> -t <TOOLCHAIN> --flash
+    ```
+
+    * Mbed CLI 1
+
+    ```bash
+    $ mbed compile -m <TARGET> -t <TOOLCHAIN> --flash
+    ```
+
+Your PC may take a few minutes to compile your code.
 
 ## Testing
 
@@ -219,10 +234,17 @@ For devices with small memory, we recommend using release profiles for building 
 
 Examples:
 
+
+* Mbed CLI 2
+
+```bash
+$ mbed-tools compile -m <TARGET> -t <TOOLCHAIN> --profile release
 ```
-$ mbed export -m KW24D -i make_iar --profile release
-OR
-$ mbed compile -m KW24D -t IAR --profile release
+
+* Mbed CLI 1
+
+```bash
+$ mbed compile -m <TARGET> -t <TOOLCHAIN> --profile release
 ```
 
 ## Troubleshooting

